src/plugins/data_view_editor/public/components/data_view_editor_flyout_content.tsx

```diff
@@ -186,13 +195,50 @@ const IndexPatternEditorFlyoutContentComponent = ({
   const isLoadingDataViewNames$ = useRef(new BehaviorSubject<boolean>(true));
   const existingDataViewNames$ = useRef(new BehaviorSubject<string[]>([]));
   const isLoadingDataViewNames = useObservable(isLoadingDataViewNames$.current, true);
+  const newItem$ = useRef(new Subject());
+
+  // I'm pretty sure this is here for debugging purposes
+  useEffect(() => {
+    const a = dataViewEditorService.matchedIndices$
+      .pipe(distinctUntilChanged(isEqual))
+      .subscribe((item) => {
+        newItem$.current.next(item);
+        newItem$.current.next(undefined);
+      });
+
+    const b = newItem$.current.subscribe((item) => console.log('newItem$', item));
+
+    return () => {
+      a.unsubscribe();
+      b.unsubscribe();
+    };
+  }, [dataViewEditorService.matchedIndices$]);
+
+  // newItem$.subscribe(dataViewEditorService.matchedIndices$);
+
+  const indicesProvider = useCallback(async () => {
+    console.log('indicesProvider is firing!');
+    const rollupIndex = undefined;
+    /* await firstValueFrom(
+      dataViewEditorService.rollupIndex$.pipe(first((data) => data !== null))
+    );
+    */
+
+    const matchedIndicesResult = await firstValueFrom(
```

Comment:
I am not seeing in this PR the following

```
matchedIndices$.next(undefined);
```

Which seems to be required for the `indicesProvider` to return the latest data. WDYT?